### PR TITLE
Release v0.15.2

### PR DESCRIPTION
**Pull request by ni4**
As title says.

### CODE REVIEW

**Comment by antonsviridenko** [comment]
I just want to remind about https://github.com/rnpgp/rnp/issues/1513 
and suggest that we start publishing checksums of release tarballs, signatures and corresponding public keys for verification
starting from this release.

**Review by antonsviridenko — APPROVED** [review]

**Comment by ni4** [comment]
> I just want to remind about #1513
> and suggest that we start publishing checksums of release tarballs, signatures and corresponding public keys for verification
> starting from this release.

Some documentations and script added here: https://github.com/rnpgp/release-sign/pull/1

**Comment by ni4** [comment]
@ronaldtse @rrrooommmaaa ping for review. Looks like we have a good chance to finally merge this, before the next wave of mystical CI failures :-)

**Review by ronaldtse — APPROVED** [review]
Thanks for the fantastic work @ni4 !

**Comment by ni4** [comment]
Thanks, @ronaldtse ! As it is release, let's wait for @rrrooommmaaa 's approval.

File: src/lib/pgp-key.cpp [hunk 1]
```diff
@@ -1439,27 +1453,15 @@ pgp_key_t::valid_till_common(bool expiry) const
...
-    uint64_t till = primary.valid_till();
-    /* if primary key was never valid then subkey was not either */
-    if (!till) {
-        return till;
-    }
-    /* subkey cannot be valid longer then the primary key */
-    return std::min(till, valid_till_common(expired() || primary.expired()));
+    /* TODO: consider implementing more sophisticated checks, as key validity time could
+     * possibly be non-continuous */
+    return (timestamp >= creation()) && timestamp && (timestamp <= valid_till());
```
Comment (rrrooommmaaa) [inline src/lib/pgp-key.cpp:1464]: Have we considered y2k38 problem here? That is, MAX_INT32 != MAX_INT32 on some systems.
However, this method is probably used for validity of creation() done in the past, and people will hardly use Debian9 after year 2038....

Comment (ni4) [reply src/lib/pgp-key.cpp:1464]: @rrrooommmaaa All variables here are `uint64_t`, or do I miss something?

File: src/lib/rnp.cpp [hunk 2]
```diff
@@ -6175,12 +6168,72 @@ try {
...
+    // search pubring
+    pgp_key_t *pub = rnp_key_store_search(primary_key->ffi->pubring, &search, NULL);
+    // search secring
+    pgp_key_t *sec = rnp_key_store_search(primary_key->ffi->secring, &search, NULL);
+
+    if (!sec && keyflag != PGP_KF_ENCRYPT) {
+        return RNP_ERROR_NO_SUITABLE_KEY;
+    }
+
+    if (pub || sec) {
```
Comment (rrrooommmaaa) [inline src/lib/rnp.cpp:6223]: We can move pubring search here

Comment (ni4) [reply src/lib/rnp.cpp:6223]: @rrrooommmaaa This was extracted to this issue: https://github.com/rnpgp/rnp/issues/1570
Actually, as release only cherry-picks commits from the master branch, we could not change code here, as first we'll need to create issues/PRs for the master branch.

File: src/tests/cli.cpp [hunk 3]
```diff
@@ -601,6 +603,184 @@ TEST_F(rnp_tests, test_cli_rnpkeys)
...
+    assert_int_not_equal(key_generate(GENKEYS, "expiration_underflow_day@rnp", "2037-02-00"),
+                         0);
+    assert_int_not_equal(key_generate(GENKEYS, "expiration_underflow_month@rnp", "2037-00-01"),
+                         0);
+    assert_int_not_equal(key_generate(GENKEYS, "expiration_underflow_year@rnp", "1800-01-01"),
+                         0);
+    assert_int_not_equal(key_generate(GENKEYS, "expiration_overflow@rnp", "200y"), 0);
+    assert_int_not_equal(key_generate(GENKEYS, "expiration_past@rnp", "2021-01-01"), 0);
+
+    // these should pass and go to the keystore -- 15 primary keys and 15 subkeys
```
Comment (rrrooommmaaa) [inline src/tests/cli.cpp:729]: actually 17 primary keys and 17 subkeys in the final version

File: src/lib/rnp.cpp (hunk 2 above)
Comment (rrrooommmaaa) [reply src/lib/rnp.cpp:6223]: Sure, as you see fit

File: src/lib/pgp-key.cpp (hunk 1 above)
Comment (rrrooommmaaa) [reply src/lib/pgp-key.cpp:1464]: Here -- yes. I'm thinking how they got here -- probably extended from int32_t, where MAX_INT32 has special meaning on Debian9. Just thinking.

**Review by rrrooommmaaa — APPROVED** [review]
Apart from my comments, looking good.

File: src/tests/cli.cpp (hunk 3 above)
Comment (ni4) [reply src/tests/cli.cpp:729]: - thanks, will fix via PR #1577

**Comment by ni4** [comment]
Thanks everybody! Finally, merging this!

**Pull request merged**